### PR TITLE
Add documentation to helm search --regexp (#3927)

### LINKED ISSUE #3927: docs clarifying how helm search --regexp
see #3890 for more context
This may come in the form of docs and/or further clarification in help/flag description

### PR DESCRIPTION
**Pull request by jstoja**
Hello @michelleN,

Would that be enough ? Don't hesitate to tell me if more is needed!
Closes #3927

### CODE REVIEW

**Comment by jstoja** [comment]
Should have solved the CLA now.

**Comment by jstoja** [comment]
Should have solved the CLA now.

**Comment by jstoja** [comment]
Should have solved the CLA now.

**Comment by jstoja** [comment]
Should have solved the CLA now? :joy:

File: cmd/helm/search.go [hunk 1]
```diff
@@ -63,7 +63,7 @@ func newSearchCmd(out io.Writer) *cobra.Command {
 	}
 
 	f := cmd.Flags()
-	f.BoolVarP(&sc.regexp, "regexp", "r", false, "use regular expressions for searching")
+	f.BoolVarP(&sc.regexp, "regexp", "r", false, "use regular expressions for searching in the concatenation of multiple fields separated by '\\v' (vertical tab)")
```
Comment (bacongobbler) [inline cmd/helm/search.go]: Hey! Would you mind adding this to the description of `helm search` instead, on line 33?

Something like

```
Search reads through all of the repositories configured on the system and looks for matches.
 
Repositories are managed with 'helm repo' commands.

To look for charts with a particular name (such as stable/mysql), try searching using vertical tabs \v:

helm search --regexp '\vstable/mysql\v'
```

Let me know if that won't work for you. Thanks!

**Review by bacongobbler — CHANGES_REQUESTED** [review]

File: cmd/helm/search.go (hunk 1 above)
Comment (jstoja) [reply cmd/helm/search.go]: Definitely a better way to document it!
I used your example and just added line breaks to fit the 80 cols convention.

**Comment by jstoja** [comment]
Just tell me if you need me to squash the commits :)

**Comment by jstoja** [comment]
up ? :)

**Review by michelleN — APPROVED** [review]

**Comment by michelleN** [comment]
hey @jstoja! @bacongobbler is currently enjoying some well deserved vacation right now as well, so I'm happy to jump in here. Thank you for contributing! Would you mind squashing these PRs? After that, I'd be happy to go ahead and merge.
Cheers!

**Review by bacongobbler — APPROVED** [review]
looks much better. Thank you!

**Comment by bacongobbler** [comment]
Oops, looks like there's a merge conflict here before we can merge. Mind fixing that up? Thanks so much!

**Comment by jstoja** [comment]
hey @bacongobbler I just merged the inserted documentation for the regexp parameter.
I tried to regenerate it after rebasing from master but it removes the `kubeconfig` recently added and I don't know why.

**Review by bacongobbler — APPROVED** [review]

**Pull request merged**